lib/plek.rb

```diff
@@ -159,6 +162,24 @@ def find_uri(*args)
 
 private
 
+  # TODO: clean up call sites throughout alphagov and then delete clean_name.
+  def clean_name(service)
+    service.to_s.downcase.strip.gsub(/[^a-z.-]+/, "")
+  end
+
+  def should_prefix(name)
+    !@unprefixable_hosts.include?(name)
+  end
+
+  def should_use_http_for_domain(domain)
+    normalised_domain = domain.delete_suffix(".")
+    if normalised_domain == ""
+      @http_domains.include?(".")
```

Comment:
I'm a bit confused by these lines, are we using `.` as a wildcard? that only applies if you have an empty suffix? I'd wonder if this is a bit cryptic as I didn't catch this until reviewing the tests.

I wonder if there are some more explicit options that can be explored. We could make an assumption that hostname with only a single label is always HTTP as that's what we anticipate for now, or could have an env var of `PLEK_HTTP_SINGLE_LABEL` hostname perhaps that's a toggle?